fabric-commands-v1/src/main/java/net/fabricmc/fabric/api/command/v1/CommandRegistrationCallback.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.command.v1;
+
+import com.mojang.brigadier.CommandDispatcher;
+
+import net.minecraft.server.command.ServerCommandSource;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Callback for when a server registers all commands.
```

Comment:
Please add a brief example, it may use [...] but should suffice to get an idea how to use this as an experienced Java dev without the wiki or digging deeply into the code.

Reply:
@sfPlayer1 Mind elaborate how it is like? like `{@code CommandRegistrationCallback.EVENT.register((dedicated, dispatcher) -> SomeCommand.register(distpacher))}`?

Reply:
IMO it'd be nice to also outline the MC classes a bit, i.e. include something like `LiteralArgumentBuilder.literal("somecmd").executes(context -> [...])`

Reply:
I think `CommandManager.literal/argument` would be better than the pure Brigadier ones because those often run into some issues with generics.

Reply:
lol I guess I predicted what you were looking for.

Well I did author the command wiki page so I have some experience with this